### PR TITLE
Dilpreet | Test | Support image on variants - Version 1

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/86cxajg97

## Release Notes

- **New Features**
  - Added ability to display variant images in invoice templates.
  - Introduced file upload and preview functionality for inventory items.
  - Enhanced localization support for file handling actions.

- **Improvements**
  - Updated invoice template layouts to support variant image display.
  - Improved file attachment management in inventory and ledger sections.
  - Added dynamic image path handling for different environments.

- **Localization**
  - Added new translation strings for file upload, preview, and attachment actions in English, Hindi, and Marathi.

- **Bug Fixes**
  - Corrected method naming conventions across multiple components.

### CODE REVIEW

**Comment by dilpreetsinghofficial** [comment]
self reviewed

File: apps/web-giddh/src/app/new-inventory/component/preview-variant-image/preview-variant-image.component.html [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<div class="dialog-header">
+    <span mat-dialog-title class="font-16">{{inputData?.localeData?.preview_image}}</span>
+    <i mat-dialog-close class="icon-cross"></i>
+</div>
+<div mat-dialog-content class="dialog-body d-flex">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/preview-variant-image/preview-variant-image.component.html]:
```suggestion
<div mat-dialog-content class="dialog-body">
```

File: apps/web-giddh/src/app/new-inventory/component/preview-variant-image/preview-variant-image.component.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    <div class="w-100">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/preview-variant-image/preview-variant-image.component.html]:
```suggestion
    <figure>
```

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss [hunk 3]
```diff
@@ -480,6 +487,58 @@ table.stock-table {
...
+        &::-webkit-file-upload-button {
+            display: none;
+        }
+    }
+
+    .remove-attachment {
+        height: 25px;
+        width: 25px;
+        padding: 2px;
+        color: black;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss]: user var

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss [hunk 4]
```diff
@@ -480,6 +487,58 @@ table.stock-table {
...
+    .attachment-placeholder {
+        color: var(--color-giddh-blue);
+    }
+
+    input {
+        left: 0;
+        border-radius: 5px;
+        z-index: 1;
+        opacity: 0;
+        cursor: pointer;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss]: use class

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss [hunk 5]
```diff
@@ -480,6 +487,58 @@ table.stock-table {
     }
 }
 
+.variant-attachment {
+    height: 48px;
+
+    .attachment-add-icon {
+        width: 36px;
+        height: 36px;
+        text-align: center;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss]: Use class

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss [hunk 6]
```diff
@@ -480,6 +487,58 @@ table.stock-table {
...
+        line-height: 36px;
+        border-radius: 50%;
+        border: 1px solid var(--color-grey-bootstrap-border);
+    }
+
+    &.text-success {
+        .attachment-add-icon {
+            border-color: var(--color-success-enable);
+            font-size: var(--font-size-12);
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss]: use class

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss [hunk 7]
```diff
@@ -480,6 +487,58 @@ table.stock-table {
...
+        }
+
+        .attachment-placeholder {
+            color: var(--color-success-enable);
+            width: 60%;
+            white-space: nowrap;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss]: use class "text-ellipsis"

written in style1

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss [hunk 8]
```diff
@@ -410,6 +412,11 @@ table.stock-table {
                 min-width: 80px;
             }
 
+            &.variant-image {
+                max-width: 165px;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss]:
```suggestion
               width: clamp(165px, 165px, 165px);
```

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html [hunk 9]
```diff
@@ -4902,10 +4857,25 @@ <h1 class="section-head p-0">
...
                     (click)="deleteStock()"
+                    [disabled]="uploadAttachmentInProgress$ | async"
                 >
                     {{ commonLocaleData?.app_delete }}
                 </button>
             </div>
         </div>
     </div>
 </ng-container>
+<!-- <ng-template #previewImage>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html]: Remove this

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts [hunk 10]
```diff
@@ -247,6 +254,16 @@ export class StockCreateEditComponent implements OnInit, OnDestroy {
...
     /** Discounts list */
     public discountsList: IDiscountList[] = [];
+    /** True if file upload in progress */
+    public isFileUploading: boolean = false;
+    /** Name of the selected file */
+    public selectedFileName: string = '';
+    /** Hold variant index*/
+    public variantIndex: number;
+    /** Upload attachment in progress Observable */
+    public uploadAttachmentInProgress$: Observable<any> = this.componentStore.uploadAttachmentInProgress$;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts]:
```suggestion
    public uploadAttachmentInProgress$: Observable<boolean> = this.componentStore.uploadAttachmentInProgress$;
```

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html [hunk 11]
```diff
@@ -1399,6 +1399,28 @@ <h2>Table</h2>
                                 </ng-container>
                             </td>
                         </tr>
+                        <tr *ngIf="fieldsAndVisibility.table.showVariantImage">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html]:
```suggestion
                        <tr *ngIf="fieldsAndVisibility.table.showVariantImage && customTemplate.templateType !== 'thermal_template'">
```

File: apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html [hunk 12]
```diff
@@ -1399,6 +1399,28 @@ <h2>Table</h2>
...
+                            <td *ngIf="customTemplate.templateType !== 'thermal_template'">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/invoice/templates/edit-template/filters-container/content-filters/content.filters.component.html]:
```suggestion
                            <td>
```

File: apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/gst-template-a/gst-template-a.component.html [hunk 13]
```diff
@@ -898,6 +912,9 @@
                                     [innerHTML]='rupeeSymbol'></span>1500.00
                                <p><span [innerHTML]='dollarSymbol'></span>21.90</p>
                             </td>
+                            <td *ngIf="fieldsAndVisibility.table.showVariantImage?.display">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/invoice/templates/edit-template/out-tempate/templates/gst-template-a/gst-template-a.component.html]: remove the code which you add inside commented code

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts [hunk 14]
```diff
@@ -2204,4 +2251,153 @@ export class StockCreateEditComponent implements OnInit, OnDestroy {
...
+                this.componentStore.uploadAttachment({
+                    file: blob,
+                    fileName: selectedFile.name,
+                    type: 'variant'
+                });
+            });
+        }
+    }
+
+
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts]: remove extra line

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts [hunk 15]
```diff
@@ -2204,4 +2251,153 @@ export class StockCreateEditComponent implements OnInit, OnDestroy {
...
+    /**
+     * This will be use for delete attachment files
+     *
+     * @param {number} index
+     * @memberof StockCreateEditComponent
+     */
+    public deleteAttachment(index: number): void {
+        this.variantIndex = index;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts]: write if (index) then all code

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts [hunk 16]
```diff
@@ -2204,4 +2251,153 @@ export class StockCreateEditComponent implements OnInit, OnDestroy {
...
+        const variant = this.stockForm.variants[index];
+        const attachmentDeleteConfig = this.generalService.getAttachmentDeleteConfiguration(this.localeData, this.commonLocaleData);
+
+        const dialogRef = this.dialog.open(NewConfirmationModalComponent, {
+            width: '630px',
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.ts]:
```suggestion
             panelClass: "mat-dialog-md",
```

iski value 700px h

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html [hunk 17]
```diff
@@ -193,6 +193,41 @@
...
+                                                </div>
+                                                <div *ngIf="stockForm.variants[0].attachmentUniqueName">
+                                                    <div
+                                                        class="variant-attachment mat-form-field-border d-flex align-items-center column-gap1 p-2 text-success">
+                                                        <div class="attachment-add-icon">
+                                                            <i class="icon-tick"></i>
+                                                        </div>
+                                                        <img class="cursor-pointer" [matTooltip]="localeData?.preview_image" matTooltipPosition="above"
+                                                            height="30" width="30" [src]="stockForm.variants[0].attachmentName"
+                                                            (click)="previewVariantImage(stockForm.variants[0].attachmentUniqueName)" alt="variantImage" />
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html]:
```suggestion
                                                            (click)="previewVariantImage(stockForm.variants[0].attachmentUniqueName)" alt="Variant Image" />
```

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html [hunk 18]
```diff
@@ -193,6 +193,41 @@
...
+                                                        <button mat-icon-button (click)="deleteAttachment(0)" class="ml-auto remove-attachment font-16">
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html]:
```suggestion
                                                        <button mat-icon-button (click)="deleteAttachment(0)" class="ml-auto remove-attachment">
```

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html [hunk 19]
```diff
@@ -193,6 +193,41 @@
...
+                                                            <i class="icon-cross"></i>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html]:
```suggestion
                                                            <i class="icon-cross font-16"></i>
```

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html [hunk 20]
```diff
@@ -1109,573 +1147,490 @@ <h1 class="section-head p-0">
...
+                                                                            <div class="variant-attachment mat-form-field-border d-flex align-items-center column-gap1 p-2"
+                                                                                [matTooltip]="localeData?.no_file_chosen" matTooltipPosition="right">
+                                                                                <input accept="image/png, image/jpeg" [ngClass]="{'cursor-disable':uploadAttachmentInProgress$ | async}"
+                                                                                    [disabled]="uploadAttachmentInProgress$ | async" type="file" class="position-absolute"
+                                                                                    name="variantFile{{ i }}" id="variantFile{{ i }}" (change)="uploadFile($event, i)" />
+                                                                                <ng-container *ngIf="variant.isUploading">
+                                                                                    <mat-spinner [diameter]="30"></mat-spinner>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html]:
```suggestion
                                                                                    <mat-spinner *ngIf="variant.isUploading" [diameter]="30"></mat-spinner>
```

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html [hunk 21]
```diff
@@ -1109,573 +1147,490 @@ <h1 class="section-head p-0">
...
+                                                                                </ng-container>
+                                                                                <ng-container *ngIf="!variant.isUploading">
+                                                                                    <div class="attachment-add-icon">
+                                                                                        <i class="icon-add"></i>
+                                                                                    </div>
+                                                                                    <span class="attachment-placeholder">{{ localeData?.attach_file || 'Attach File' }}</span>
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.html]:
```suggestion
                                                                                    <span class="attachment-placeholder">{{ localeData?.attach_file }}</span>
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss [hunk 22]
```diff
@@ -480,6 +486,54 @@ table.stock-table {
...
 
+.variant-attachment {
+    height: 48px;
+
+    .attachment-add-icon {
+        width: 36px;
+        height: 36px;
+        text-align: center;
+        line-height: 36px;
+        border-radius: 50%;
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/new-inventory/component/stock-create-edit/stock-create-edit.component.scss]: rounded-circle

**Review by dvCodeWorld — APPROVED** [review]

**Pull request merged**